utility/src/main/java/org/oppia/android/util/logging/performancemetrics/ProdPerformanceMetricsUtils.kt

```diff
@@ -0,0 +1,88 @@
+package org.oppia.android.util.logging.performancemetrics
+
+import android.app.ActivityManager
+import android.content.Context
+import android.net.TrafficStats
+import org.oppia.android.app.model.OppiaMetricLog
+import org.oppia.android.util.logging.ConsoleLogger
+import java.io.File
+import javax.inject.Inject
+
+/** Utility to extract performance metrics from the underlying Android system. */
+class ProdPerformanceMetricsUtils @Inject constructor(
+  private val context: Context,
+  private val consoleLogger: ConsoleLogger
+) : PerformanceMetricsUtils {
+
+  override fun getApkSize(): Long {
+    var apkSize: Long = 0
+    try {
+      val apkPath =
+        context.packageManager.getPackageInfo(context.packageName, 0).applicationInfo.sourceDir
+      val apkFile = File(apkPath)
+      apkSize = ((apkFile.length() / 1024))
+    } catch (e: Exception) {
```

Comment:
We should avoid catching exceptions for control flow. Is there any way to check state to avoid needing to catch here?

Ditto below.

Reply:
understood -- done.